src/libpriv/rpmostree-core.c

```diff
@@ -2860,21 +2861,46 @@ handle_file_dispositions (RpmOstreeContext *self,
   return TRUE;
 }
 
+typedef struct {
+  GHashTable *files_skip;
+  GPtrArray  *files_remove_regex;
+} FilterData;
+
 static OstreeRepoCheckoutFilterResult
 checkout_filter (OstreeRepo         *self,
                  const char         *path,
                  struct stat        *st_buf,
                  gpointer            user_data)
 {
-  GHashTable *files_skip = user_data;
-  if (g_hash_table_contains (files_skip, path))
-    return OSTREE_REPO_CHECKOUT_FILTER_SKIP;
+  GHashTable *files_skip = ((FilterData*)user_data)->files_skip;
+  GPtrArray *files_remove_regex = ((FilterData*)user_data)->files_remove_regex;
+
+  if (files_skip && g_hash_table_size (files_skip) > 0)
+    {
+      if (g_hash_table_contains (files_skip, path))
+        return OSTREE_REPO_CHECKOUT_FILTER_SKIP;
+    }
+
+  if (files_remove_regex)
+    {
+      for (guint i = 0; i < files_remove_regex->len; i++)
+        {
+          GRegex *regex = g_ptr_array_index (files_remove_regex, i);
+          if (g_regex_match (regex, path, 0, NULL))
+            {
+              g_print ("Skipping file %s from checkout\n", path);
```

Comment:
Leftover debugging?

Reply:
This was from the original `handle_remove_file_from_packages()`'s 
```
if (g_regex_match (regex, file, 0, NULL))	
                {	
                  if (file[0] == '/')	
                    file++;	

                  g_print ("Deleting: %s\n", file);	
                  if (!glnx_shutil_rm_rf_at (rootfs_fd, file, cancellable, error))	
                    return FALSE;
```

Reply:
Ahh right OK. Normally, we use `rpmostree_output_message` in the core since that code might be used as part of a client <-> daemon transaction or directly via composes. I think it's fine like this for now and isn't worth a respin, since the client-side doesn't use this functionality anyway. So let's do that in a follow-up!